pymtl3/passes/rtlir/behavioral/BehavioralRTLIRGenL1Pass.py

```diff
@@ -220,17 +220,32 @@ def visit_Attribute( s, node ):
   def visit_Subscript( s, node ):
     value = s.visit( node.value )
     if isinstance( node.slice, ast.Slice ):
-      if not node.slice.step is None:
+      if node.slice.step is not None:
         raise PyMTLSyntaxError( s.blk, node,
           'Slice with steps is not supported!' )
       lower, upper = s.visit( node.slice )
       ret = bir.Slice( value, lower, upper )
       ret.ast = node
       return ret
 
+    # signal[ index ]
+    # index might be a slice object!
     if isinstance( node.slice, ast.Index ):
       idx = s.visit( node.slice )
-      ret = bir.Index( value, idx )
+      # If we haev a static slice object then use it
+      if isinstance( idx, bir.FreeVar ) and isinstance( idx.obj, slice ):
+        slice_obj = idx.obj
+        if slice_obj.step is not None:
+          raise PyMTLSyntaxError( s.blk, node,
+            'Slice with steps is not supported!' )
```

Comment:
@cbatten do you think it is useful in RTL to support slice with steps? Are there cases where gathering every odd bit is part of the design .. ? It might be useful for hardware random number generators? Oh well if they want to implement it int the current DSL they could just connect every other bit to a half-width wire using a loop with s.connect ..